cpp/src/gandiva/encrypt_utils.cc

```diff
@@ -0,0 +1,83 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "encrypt_utils.h"
+
+#include "gandiva/gdv_function_stubs.h"
+
+namespace gandiva {
+
+int32_t aes_encrypt(int64_t context, unsigned char* plaintext, int32_t plaintext_len,
```

Comment:
This is going to perform poorly. This function is called for each and every row in the query.

This has too much overhead - please talk to Projjal about how to implement this correctly

Reply:
This is ok for now